src/HIVCohortFilter/HIVCohortCases.jsx

```diff
@@ -0,0 +1,1205 @@
+                      rightIcon='download'
+                      id='download-control-button'
+                      enabled={!this.state.inLoadingState
+                        && this.state.resultAlreadyCalculated
+                        && this.state.subjectControl.length > 0}
+                      onClick={this.downloadControl}
+                      buttonType='secondary'
+                      isPending={this.state.inLoadingState}
+                    />
+                  </React.Fragment>
+                }
+              </div>
+            </div>
+          </div>
+        </div>
+      </React.Fragment>
+    );
+  }
+}
+
+class ECCase extends HIVCohortFilterCase {
+  /*
+  * EC Case:
+  * Below is the full algorithm description from https://ctds-planx.atlassian.net/browse/PXP-2892
+  * The UI displays a 'decision tree' (just hardcoded svg), and makes an es query based on
+  * sliding window size and viral load number.
+  * Definitions:
+  * - thrpy is the treatment that patient used since the last visit ( follow_up)
+  * - thrpyv is the treatment that patient uses at the visit ( follow_up )
+  * Definitions:
+  * - Never received HAART treatment: follow_up.thrpyv != HAART
+  * - viral load< X: followup.viral_load < X
+  * - Consecutive Y month: (last_followup.visit_number - first_followup.visit_number)*6 = Y
+  * If there are missing visit number, eg: patient has visit_number 1, 3, 4, 5. 
+  * Just consider the missing one still maintain the same viral load)
+  * The EC criteria are:
+  * - Patients' hiv_status are positive, have never received HAART treatment and
+  * have consecutive follow ups for Y months with viral load < X
+  * - User will type in the time period Y and viral load threshold X, and the app will
+  * show the count for EC case and control subjects. And have buttons to download
+  * the clinical manifest for them.
+  */
+  constructor(props) {
+    super(props);
+    this.state = {
+      viralLoadFromUser: undefined,
+      numConsecutiveMonthsFromUser: undefined,
+      subjectNeither: [],
+      subjectEC: [],
+      subjectControl: [],
+      inLoadingState: false,
+      isReadyToCalculate: false,
+      resultAlreadyCalculated: false,
+    };
+    this.downloadEC = this.downloadEC.bind(this);
+    this.makeCohortJSONFile = this.makeCohortJSONFile.bind(this);
+    this.checkReadyToCalculate = this.checkReadyToCalculate.bind(this);
+    this.viralLoadInputRef = React.createRef();
+    this.numConsecutiveMonthsInputRef = React.createRef();
+    this.showCount = this.showCount.bind(this);
+  }
+
+  getBucketByKeyWithHAARTVAR(bucketKey, isHAART) {
+    // The below query differs from the PTC case in that there is no viral_load check.
+    // This is because the EC and LTNP cases uses this function to find people who have
+    // never received haart treatments; we need to look at *all* their followups.
+    // (Read the function getBucketByKey() defined in the HIVCohortFilterCase class.)
+    const query = `
```

Comment:
This is just code structure and refactoring: since we're having more and more queries it'd be good to separate requests and UI rendering apart, how about make those queries into another util files for UI component to import?

Reply:
This is true! But I'd argue this particular refactor would make the code less readable -- the algorithm logic is easier to follow if the query is in the same place. After splitting the three cases into 3 files, each file is under 400 lines, i feel like that's an appropriate file length already?